web3.js/src/programs/address-lookup-table/state.ts

```diff
@@ -0,0 +1,84 @@
+import * as BufferLayout from '@solana/buffer-layout';
+
+import assert from '../../utils/assert';
+import * as Layout from '../../layout';
+import {PublicKey} from '../../publickey';
+import {u64} from '../../utils/bigint';
+import {decodeData} from '../../account-data';
+
+export type AddressLookupTableState = {
+  deactivationSlot: bigint;
+  lastExtendedSlot: number;
+  lastExtendedSlotStartIndex: number;
+  authority?: PublicKey;
+  addresses: Array<PublicKey>;
+};
+
+export type AddressLookupTableAccountArgs = {
+  key: PublicKey;
+  state: AddressLookupTableState;
+};
+
+/// The serialized size of lookup table metadata
+const LOOKUP_TABLE_META_SIZE = 56;
+
+export class AddressLookupTableAccount {
+  key: PublicKey;
+  state: AddressLookupTableState;
+
+  constructor(args: AddressLookupTableAccountArgs) {
+    this.key = args.key;
+    this.state = args.state;
+  }
+
+  isActive(): boolean {
+    const U64_MAX = 2n ** 64n - 1n;
```

Comment:
Fun fact: this change made `web3.js` incompatible with React Native because:

* JavaScript Core (JSC) doesn't support the exponentiation operator (`**`) so there's a transform that converts `2n ** 64n` to `Math.pow(2n, 64n)` and _then_ you're in trouble because `Math.pow()` doesn't work with `bigint` (runtime fatal)
* Hermes supports the exponentiation operator but doesn't support `bigint` yet.

No change necessary – I'm working on it.

Reply:
https://twitter.com/steveluscher/status/1563940218444840960

Reply:
Ouch, thanks for the heads up. Seems worthwhile to use a literal here then, or do you think it's not worth the trouble?

Reply:
There are now three places in `web3.js` that use `BigInt`

* This, with exponentiation
* @noble/ed25519, with exponentiation
* @noble/secp256k1, with exponentiation

If we inline the computation _and_ convince @noble to do the same (eg. just literally change this to `BigInt('18446744073709551616')` and comment it) then:

* Less CPU overhead (nice)
* JSC: still fucked because JSC doesn't support `BigInt`
* JSC with `BigInt` polyfill: still fucked because polyfills don't permit arithmetic (`require('big-integer')(1) + require('big-integer')(1)` is actually nonsense, because it's basically `Object + Object`)
* Pre 0.70 Hermes: still fucked because of no `BigInt` support
* \>=0.70 Hermes: Works, but so does the original exponentiated code.

I'm going to post about this on September 1st, but my recommendation going forward for anyone who wants to use Solana libraries in React Native going forward will be to upgrade to React Native 0.70 and switch to Hermes. Big integers are critical for most crypto use cases, Hermes is now the _default_ in React Native from 0.70 onward, and the way out of this mess is forward.

Reply:
Ok, really appreciate the breakdown on this. I'll leave as-is then